internals/daemon/api_checks.go

```diff
@@ -66,3 +67,34 @@ func v1GetChecks(c *Command, r *http.Request, _ *UserState) Response {
 	}
 	return SyncResponse(infos)
 }
+
+func v1PostChecks(c *Command, r *http.Request, _ *UserState) Response {
+	var payload struct {
+		Action string `json:"action"`
+		Check  string `json:"check"`
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode data from request body: %v", err)
+	}
+	switch payload.Action {
+	case "run":
+		if payload.Check == "" {
+			return BadRequest("no check to %s provided", payload.Action)
+		}
+
+		check, ok := c.d.overlord.PlanManager().Plan().Checks[payload.Check]
+		if !ok {
+			return SyncResponse("check not found")
+		}
+
+		err := c.d.overlord.CheckManager().RunCheck(r.Context(), check)
```

Comment:
Again, let's split onto two lines so it's easier to see what's being called. Similar code in v1GetChecks:

```go
	checkMgr := c.d.overlord.CheckManager()
	checks, err := checkMgr.Checks()
	if err != nil {
		return InternalError("%v", err)
	}
```